backend/application/syson-application/src/test/java/org/eclipse/syson/application/controllers/diagrams/general/view/GVTopNodeCreationTests.java

```diff
@@ -0,0 +1,211 @@
+        this.verifier = StepVerifier.create(flux);
+        this.diagram = this.givenDiagram.getDiagram(this.verifier);
+        this.diagramDescription = this.givenDiagramDescription.getDiagramDescription(SysMLv2Identifiers.GENERAL_VIEW_EMPTY_PROJECT,
+                SysMLv2Identifiers.GENERAL_VIEW_DIAGRAM_DESCRIPTION_ID);
+        this.diagramDescriptionIdProvider = new DiagramDescriptionIdProvider(this.diagramDescription, this.diagramIdProvider);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        if (this.verifier != null) {
+            this.verifier.thenCancel()
+                    .verify(Duration.ofSeconds(10));
+        }
+    }
+
+    @Sql(scripts = { "/scripts/syson-test-database.sql" }, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = { "/scripts/cleanup.sql" }, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD, config = @SqlConfig(transactionMode = SqlConfig.TransactionMode.ISOLATED))
+    @ParameterizedTest
+    @MethodSource("topNodeParameters")
+    public void createTopNode(EClass eClass, int compartmentCount) {
+        String creationToolId = this.diagramDescriptionIdProvider.getDiagramCreationToolId(this.descriptionNameGenerator.getCreationToolName(eClass));
+
+        this.verifier.then(() -> this.nodeCreationTester.createNodeOnDiagram(SysMLv2Identifiers.GENERAL_VIEW_EMPTY_PROJECT,
+                this.diagram,
+                creationToolId));
+
+        Consumer<DiagramRefreshedEventPayload> updatedDiagramConsumer = payload -> Optional.of(payload)
+                .map(DiagramRefreshedEventPayload::diagram)
+                .ifPresentOrElse(newDiagram -> {
+                    List<Node> newNodes = this.diagramComparator.newNodes(this.diagram.get(), newDiagram);
+                    List<Edge> newEdges = this.diagramComparator.newEdges(this.diagram.get(), newDiagram);
+                    int createdNodesExpectedCount = 1 + compartmentCount;
+                    assertThat(newNodes).as("The diagram should contain " + createdNodesExpectedCount + " new node").hasSize(createdNodesExpectedCount);
+                    assertThat(newEdges).as("The diagram should contain " + 0 + " new edge").hasSize(0);
+
+                    String newNodeDescriptionName = this.descriptionNameGenerator.getNodeName(eClass);
+                    Optional<Node> optNewNode = newNodes.stream()
+                            .filter(n -> Objects.equals(n.getDescriptionId(), this.diagramDescriptionIdProvider.getNodeDescriptionId(newNodeDescriptionName)))
+                            .findFirst();
+                    assertThat(optNewNode).as("The diagram should contain a new node with " + newNodeDescriptionName + " description");
+                    Node newNode = optNewNode.get();
+                    assertThat(newNode.getChildNodes()).as("The new node should contain " + compartmentCount + " compartments").hasSize(compartmentCount);
```

Comment:
Part of these checks could be factorized in a later iteration (once more kind of tests are implemented and we better know what should be factorized)